crates/swc_ecma_parser/src/parser/stmt.rs

```diff
@@ -799,6 +803,14 @@ impl<'a, I: Tokens> Parser<I> {
                 // typescript allows `declare` vars not to have initializers.
                 if self.ctx().in_declare {
                     None
+                } else if *kind == VarDeclKind::Const && self.ctx().strict {
+                    let syntax_error = if self.syntax().typescript() {
+                        SyntaxError::TS1155
+                    } else {
+                        SyntaxError::ConstDeclarationsRequireInitialization
+                    };
+
+                    syntax_error!(self, span!(self, start), syntax_error)
```

Comment:
I think we can recover the error here.

`self.emit_err()` can be used to proceed parsing while reporting an error.

Reply:
resolved in [48d5806](https://github.com/swc-project/swc/pull/3742/commits/48d58060388df760ca6904bf91148ee858211565)